addons/pos_razorpay/models/pos_payment_method.py

```diff
@@ -0,0 +1,123 @@
+import logging
+import requests
+
+from odoo.exceptions import UserError
+from odoo import fields, models, api, _
+
+_logger = logging.getLogger(__name__)
+REQUEST_TIMEOUT = 30
+BASE_URL = {
+    'test': 'https://demo.ezetap.com/api/3.0/p2padapter/',
+    'production': 'https://www.ezetap.com/api/3.0/p2padapter/',
+}
+
+class PosPaymentMethod(models.Model):
+    _inherit = 'pos.payment.method'
+
+    razorpay_tid = fields.Char(string='Razorpay Device Serial No', help="Device Serial No \n ex: 7000012300")
+    razorpay_allowed_payment_modes = fields.Selection(selection=[('all', 'All'), ('card', 'Card'), ('upi', 'UPI'), ('bharatqr', 'BHARATQR')], default='all', help="Choose allow payment mode: \n All/Card/UPI or QR")
+    razorpay_username = fields.Char(string="Razorpay Username", help="Username(Device Login) \n ex: 1234500121")
+    razorpay_api_key = fields.Char(string="Razorpay API Key", help="API Key \n ex: abcdefgh-0000-1a23-ab00-12a3b4cd5ef6")
+    razorpay_test_mode = fields.Boolean(string="Razorpay Test Mode", default=False, help="Turn it on when in Test Mode")
+
+    def _get_payment_terminal_selection(self):
+        return super()._get_payment_terminal_selection() + [('razorpay', 'Razorpay')]
+
+    def _call_razorpay(self, endpoint, payload=None):
+        """ Make a request to Razorpay API.
+
+        :param str endpoint: The endpoint to be reached by the request.
+        :param dict payload: The payload of the request.
+        :return The JSON-formatted content of the response.
+        :rtype: dict
+        """
+        try:
+            api_url = BASE_URL['test'] if self.razorpay_test_mode else BASE_URL['production']
+            response = requests.post(api_url+endpoint, json=payload, timeout=REQUEST_TIMEOUT)
```

Comment:
In your case it is better to use a session [1] to build your request.
Why:
When you use `requests.post()` it create each time a new underlying TCP connection.
With a creation of an instance of requests.session you can reuse it and use persistent cookies across all requests [2]. Which can result in a significant performance increase.
Morever you set your headers or body constant only onetime, in your case it can reduce your number of lines 

[1] https://requests.readthedocs.io/en/latest/user/advanced/
[2] https://en.wikipedia.org/wiki/HTTP_persistent_connection